src/EnergyPlus/SingleDuct.cc

```diff
@@ -6235,6 +6242,16 @@ namespace SingleDuct {
         }
     }
 
+    void SingleDuctAirTerminal::CalcOutdoorAirVolumeFlowRate()
+    {
+        // calculates the amount of outdoor air volume flow rate using the supply air flow rate and OA fraction
+        if (this->AirLoopNum > 0) {
+            this->OutdoorAirFlowRate = (this->sd_airterminalOutlet.AirMassFlowRate / StdRhoAir) * DataAirLoop::AirLoopFlow(this->AirLoopNum).OAFrac;
+        } else {
+            // do nothing for now
+        }
+    }
+
```

Comment:
This is the common new member function added to singleduct air terminal struct. This is used for all single air terminal but currently only four of the single duct air terminals have this report variable.

Reply:
"Same" function in two locations. I'm assuming there's not a parent component. Should there (eventually) be one?